xarray/tests/test_datatree.py

```diff
@@ -1534,6 +1550,18 @@ def test_binary_op_on_datatree(self):
         result = dt * dt  # type: ignore[operator]
         assert_equal(result, expected)
 
+    def test_arithmetic_inherited_coords(self):
+        tree = DataTree(xr.Dataset(coords={"x": [1, 2, 3]}))
```

Comment:
I think we should test the behaviour with `map_over_subtree` explicitly, because the arithmetic behaviour is merely downstream of that. Even just mapping a lambda function that returns its argument unchanged should display this bug.

Reply:
done